src/tools/r2rdump/CoreDisTools.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Reflection.PortableExecutable;
+using System.Runtime.InteropServices;
+
+namespace R2RDump
+{
+    class CoreDisTools
+    {
+        public enum TargetArch
+        {
+            Target_Host, // Target is the same as host architecture
+            Target_X86,
+            Target_X64,
+            Target_Thumb,
+            Target_Arm64
+        };
+
+        [DllImport(
+        "coredistools.dll",
+        CharSet = CharSet.Ansi,
+        CallingConvention = CallingConvention.StdCall)]
+        [return: MarshalAs(UnmanagedType.I8)]
+        public static extern long InitDisasm(TargetArch Target);
+
+        [DllImport(
+        "coredistools.dll",
+        CharSet = CharSet.Ansi,
+        CallingConvention = CallingConvention.StdCall)]
+        public static extern void DumpCodeBlock(long Disasm, ulong Address, IntPtr Bytes, int Size);
+
+        [DllImport(
+        "coredistools.dll",
+        CharSet = CharSet.Ansi,
+        CallingConvention = CallingConvention.StdCall)]
+        public static extern void FinishDisasm(long Disasm);
+
+        public unsafe static void DumpCodeBlock(long Disasm, int Address, int Offset, byte[] image, int Size)
+        {
+            fixed (byte* p = image)
+            {
+                IntPtr ptr2 = (IntPtr)(p + Offset);
+                DumpCodeBlock(Disasm, (ulong)Address, ptr2, Size);
+            }
+        }
+
+        public static long GetDisasm(Machine machine)
+        {
+            TargetArch target = TargetArch.Target_Host;
+            switch (machine)
+            {
+                case Machine.IA64:
+                case Machine.Amd64:
+                    target = TargetArch.Target_X64;
+                    break;
+                case Machine.I386:
+                    target = TargetArch.Target_X86;
+                    break;
+                case Machine.Arm:
```

Comment:
Should this be `Arm64`?

Reply:
Arm64 wasn't an option in the Machine enum. Would Machine.Arm be fore the 64-bit architecture?

Reply:
It is there in the current version: https://github.com/dotnet/corefx/blob/c45c7a7439ab2925d1a6e17e80f396a6bd35c17b/src/System.Reflection.Metadata/ref/System.Reflection.Metadata.cs#L3145

You may need to explicitly reference a higher version of System.Reflection.Metadata package in your csproj file; or bump the netcoreapp version in the csproj file to netcoreapp2.1 that should have it.